20-red-knob-sea-star/api_definitions/rest/claims_repository.yaml

```diff
@@ -0,0 +1,367 @@
+        - name: claim_id
+          in: path
+          required: true
+          schema:
+            title: Internal claim ID
+            type: string
+      responses:
+        '204':
+          description: User claim was successfully deleted.
+        '404':
+          description: The user claim was not found.
+        '422':
+          description: Validation error in submitted user or claim ID.
+
+    patch:
+      operationId: patch_user
+      summary: Revoke an existing user claim
+      description: Endpoint used to revoke a claim for a specified user.
+      tags:
+        - claims
+      parameters:
+        - name: user_id
+          in: path
+          required: true
+          schema:
+            title: Internal user ID
+            type: string
+        - name: claim_id
+          in: path
+          required: true
+          schema:
+            title: Internal claim ID
+            type: string
+        - name: revocation_date
+          in: query
+          required: true
+          schema:
+            title: Revocation Date
+            type: string
+            format: date-time
+      responses:
+        '204':
+          description: User claim was successfully saved.
+        '404':
+          description: The user was not found.
+        '422':
+          description: Validation error in submitted user data.
+
+components:
+
+  schemas:
+
+    AuthorityLevel:
+      title: AuthorityLevel
+      enum:
+        - self
+        - peer
+        - system
+        - so
+        - dac
+      type: string
+      description: Type of asserting authority
+
+    Claim:
+      title: Claim
+      required:
+        - user_id
+        - visa_type
+        - visa_value
+        - assertion_date
+        - valid_from
+        - valid_until
+        - source
+        - id
+      type: object
+      properties:
+        user_id:
+          title: ID
+          type: string
+          description: Internally used ID of the user
+        visa_type:
+          title: Visa type
+          type: string
+        visa_value:
+          title: Scope of the claim depending of the visa type
+          type: string
+        assertion_date:
+          title: Assertion date
+          type: string
+          format: date-time
+        revocation_date:
+          title: If revoked, date of revocation
+          type: string
+          format: date-time
+          nullable: true
+        valid_from:
+          title: Start date of validity
+          type: string
+          format: date-time
+        valid_until:
+          title: End date of validity
+          type: string
+          format: date-time
+        source:
+          title: Asserting organization
+          maxLength: 2083
+          minLength: 1
+          type: string
+          format: uri
+        sub_source:
+          title: Asserting sub-organization
+          maxLength: 2083
+          minLength: 1
+          type: string
+          format: uri
+          nullable: true
```

Comment:
Would this be the identifier of the DAC making the assertion?

Reply:
Yes, or maybe the hub if it is an assertion made by us. The visa only have a single source attribute, which then would contain both the organization and the sub-organization (e.g. by using a subdomain), or contain only one of them. But in the repository it may be good to have them separately.